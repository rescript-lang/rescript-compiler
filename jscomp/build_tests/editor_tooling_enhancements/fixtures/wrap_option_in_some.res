switch Some(1) {
| 1 => ()
| _ => ()
}
