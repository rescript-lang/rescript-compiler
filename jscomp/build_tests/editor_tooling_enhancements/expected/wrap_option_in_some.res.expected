
  [1;31mWe've found a bug for you![0m
  [36m/.../fixtures/wrap_option_in_some.res[0m:[2m2:3[0m

  1 [2mâ”‚[0m switch Some(1) {
  [1;31m2[0m [2mâ”‚[0m | [1;31m1[0m => ()
  3 [2mâ”‚[0m | _ => ()
  4 [2mâ”‚[0m }

  This pattern matches values of type [1;31mint[0m
  but a pattern was expected which matches values of type [1;33moption<int>[0m
  
  The value you're pattern matching on here is wrapped in an [1;33moption[0m, but you're trying to match on the actual value.
  Wrap the highlighted pattern in [1;33mSome()[0m to make it work.
=== CODE ACTIONS ===
[{"title": "Wrap in `Some()`", "kind": "quickfix" "loc": {"start": {"line": 2, "col": 19},"end": {"line": 2, "col": 20}}, "type": "wrapWith", "wrapLeft": "Some(", "wrapRight": ")"}]